common/djangoapps/third_party_auth/models.py

```diff
@@ -421,3 +427,60 @@ def current(cls, entity_id):
 
         cache.set(cls.cache_key_name(entity_id), current, cls.cache_timeout)
         return current
+
+
+class LTIProviderConfig(ProviderConfig):
+    """
+    Configuration required for this edX instance to act as a LTI
+    Tool Provider and allow users to authenticate and be enrolled in a
+    course via third party LTI Tool Consumers.
+    """
+    prefix = 'lti'
+    backend_name = 'lti'
+    icon_class = None  # This provider is not visible to users
+    secondary = False  # This provider is not visible to users
+    accepts_logins = False  # LTI login cannot be initiated by the tool provider
+    KEY_FIELDS = ('lti_consumer_key', )
+
+    lti_consumer_key = models.CharField(
+        max_length=255,
+        unique=True,
+        help_text=(
+            'The name that the LTI Tool Consumer will use to identify itself'
```

Comment:
There's no character set restriction as far as I know. I just logged in from the TC reference implementation using a consumer key of "test-consumer" so hyphens seem to work for me. Which validation criteria are failing?